data-processing/common/commands/database.py

```diff
@@ -1,11 +1,39 @@
 from abc import ABC
 from argparse import ArgumentParser
-from typing import Any
+from enum import Enum
+import logging
+from typing import Any, Optional
 
 from common.commands.base import ArxivBatchCommand, I, R
 from common.models import setup_database_connections
 
 
+class OutputForm(Enum):
+    DB = "db"
+    FILE = "file"
+    BOTH = "both"
+
+
+class OutputDetails:
+    def __init__(self, output_form: str, output_dir: Optional[str]):
+        OutputDetails.validate(output_form=output_form, output_dir=output_dir)
+        self.output_form = OutputForm(output_form)
+        self.output_dir = output_dir
+
+    @staticmethod
+    def validate(output_form: str, output_dir: Optional[str]) -> None:
+        msg = "If the output form is 'file' or 'both', an output dir must also be specified."
+        cond = (output_form in [OutputForm.FILE.value, OutputForm.BOTH.value]) == \
+            (output_dir is not None)
+        assert cond, msg
+
+    def save_to_db(self) -> bool:
```

Comment:
Style nit pick: could the name be modified to suggest that it is a check on whether a save can be performed, rather than performing that save (e.g., `can_save_to_db`)?

Reply:
Yes, thanks for the suggestion!